<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Explore KEGG profile • rbims</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Explore KEGG profile">
<meta property="og:description" content="rbims">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:creator" content="@MirnaVRL">
<meta name="twitter:site" content="@MirnaVRL">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-000000-01"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-000000-01');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rbims</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Create_Interpro_profile.html">Create Interpro profile</a>
    </li>
    <li>
      <a href="../articles/Create_KEGG_profile.html">Create KEGG profile</a>
    </li>
    <li>
      <a href="../articles/Explore_InterProScan_profile.html">Explore InterProScan profile</a>
    </li>
    <li>
      <a href="../articles/Explore_KEGG_profile.html">Explore KEGG profile</a>
    </li>
    <li>
      <a href="../articles/Installation.html">Installation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mirnavazquez/RbiMs/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Explore_KEGG_profile_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Explore KEGG profile</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mirnavazquez/RbiMs/blob/master/vignettes/Explore_KEGG_profile.Rmd"><code>vignettes/Explore_KEGG_profile.Rmd</code></a></small>
      <div class="hidden name"><code>Explore_KEGG_profile.Rmd</code></div>

    </div>

    
    
<p>First, load the rbims package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mirnavazquez/RbiMs">rbims</a></span><span class="op">)</span></code></pre></div>
<p>The second thing to do would be to read the metadata file and the <a href="https://github.com/mirnavazquez/RbiMs/blob/main/inst/extdata/KEGG_bins.txt">KofamKOALA/KofamScan</a> output. The metadata file is a tab-separated file containing the name of your bins and any extra information you would like to use for visualization.</p>
<div id="read-metadata" class="section level1">
<h1 class="hasAnchor">
<a href="#read-metadata" class="anchor"></a>Read metadata</h1>
<p>The metadata file contains external information of each bin, like sample site or taxonomy. The name of the bins in the first column is mandatory.</p>
<p>The metadata table can be read in various formats (csv, tsv, txt, xlsx); you will need to use the corresponding function to read the type of file you have. In this case, the example table of rbims is in excel format; therefore, to read the metadata, we will use the function <code>read_excel</code> from the package <code>readxl</code>. You can download the metadata example file <a href="https://github.com/mirnavazquez/RbiMs/blob/main/inst/extdata/metadata.xlsx">metadata</a> and try.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org">readxl</a></span><span class="op">)</span>
<span class="va">metadata</span><span class="op">&lt;-</span><span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html">read_excel</a></span><span class="op">(</span><span class="st">"metadata.xlsx"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 10</span>
<span class="co">#&gt;   Bin_name Sample_site  Clades  Domain  Phylum  Class Order Family Genus Species</span>
<span class="co">#&gt;   &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;  </span>
<span class="co">#&gt; 1 Bin_10   Sediment     Clade_1 d__Bac… p__Pro… c__G… o__P… f__Sa… g__O… s__Ole…</span>
<span class="co">#&gt; 2 Bin_12   Sediment     Clade_2 d__Bac… p__Pro… c__G… o__P… f__Ol… g__M… s__Mar…</span>
<span class="co">#&gt; 3 Bin_56   Sediment     Clade_3 d__Bac… p__Pro… c__G… o__P… f__Sa… g__T… s__Tha…</span>
<span class="co">#&gt; 4 Bin_113  Water_column Clade_1 d__Bac… p__Pro… c__A… o__R… f__Rh… g__P… s__Par…</span>
<span class="co">#&gt; 5 Bin_1    Water_column Clade_2 d__Bac… p__Pro… c__G… o__E… f__Al… g__A… s__Alt…</span>
<span class="co">#&gt; 6 Bin_2    Water_column Clade_3 d__Bac… p__Pro… c__G… o__E… f__Al… g__A… s__Alt…</span></code></pre></div>
<p>If you followed the create KEGG profile tutorial, you could go directly to a <a href="https://mirnavazquez.github.io/RbiMs/articles/Explore_KEGG_profile.html#example-of-exploring-an-specific-kegg-pathway-1">case example of exploring and specific pathway</a>.</p>
</div>
<div id="read-the-kegg-results" class="section level1">
<h1 class="hasAnchor">
<a href="#read-the-kegg-results" class="anchor"></a>Read the KEGG results</h1>
<p><a href="https://mirnavazquez.github.io/RbiMs/reference/read_ko.html">read_ko</a> can read multiple text files obtained from <a href="https://www.genome.jp/tools/kofamkoala/">KofamKOALA/KofamScan</a> or <a href="https://www.genome.jp/kegg/kaas/">KAAS</a>, as long as they are all in the same path in your working directory. If you use both and there are different hits for a KO in both searches, it will take the hit from KofamKOALA/KofamScan.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ko_bin_table</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/read_ko.html">read_ko</a></span><span class="op">(</span>data_kofam <span class="op">=</span><span class="st">"C:/Users/bins"</span><span class="op">)</span></code></pre></div>
</div>
<div id="map-to-the-kegg-database" class="section level1">
<h1 class="hasAnchor">
<a href="#map-to-the-kegg-database" class="anchor"></a>Map to the KEGG database</h1>
<p>Then map the KO to the rest of the features of the KEGG and rbims database.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ko_bin_mapp</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/mapping_ko.html">mapping_ko</a></span><span class="op">(</span><span class="va">ko_bin_table</span><span class="op">)</span></code></pre></div>
</div>
<div id="metabolism-subsetting" class="section level1">
<h1 class="hasAnchor">
<a href="#metabolism-subsetting" class="anchor"></a>Metabolism subsetting</h1>
<p>To explore the metabolism table, rbims has three functions to subset the table:</p>
<ul>
<li><a href="https://mirnavazquez.github.io/RbiMs/reference/get_subset_pathway.html">get_subset_pathway</a></li>
<li><a href="https://mirnavazquez.github.io/RbiMs/reference/get_subset_pca.html">get_subset_pca</a></li>
<li><a href="https://mirnavazquez.github.io/RbiMs/reference/get_subset_unique.html">get_subset_unique</a></li>
</ul>
</div>
<div id="example-of-exploring-a-specific-kegg-pathway" class="section level1">
<h1 class="hasAnchor">
<a href="#example-of-exploring-a-specific-kegg-pathway" class="anchor"></a>Example of exploring a specific KEGG pathway</h1>
<p>Let’s say that you are interested in the genes associated with the biofilm formation in Vibrio Cholerae.</p>
<ul>
<li>One option is to create a vector containing the name of the KEGG pathway associated with the biofilm formation in Vibrio Cholerae <a href="https://www.kegg.jp/pathway/map05111">map05111</a>.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Biofilm_Vibrio</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"map05111"</span><span class="op">)</span></code></pre></div>
<ul>
<li>Now, let’s extract the profile associated with that metabolic pathway.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="va">Biofilm_Vibrio_subset</span><span class="op">&lt;-</span><span class="va">ko_bin_mapp</span><span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html">drop_na</a></span><span class="op">(</span><span class="va">Pathway</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/get_subset_pathway.html">get_subset_pathway</a></span><span class="op">(</span><span class="va">Pathway</span>, <span class="va">Biofilm_Vibrio</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">Biofilm_Vibrio_subset</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 19</span>
<span class="co">#&gt;   Module Module_description      Pathway  Pathway_descripti… Cycle Pathway_cycle</span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;                   &lt;chr&gt;    &lt;chr&gt;              &lt;chr&gt; &lt;chr&gt;        </span>
<span class="co">#&gt; 1 &lt;NA&gt;   &lt;NA&gt;                    map05111 Biofilm formation… &lt;NA&gt;  &lt;NA&gt;         </span>
<span class="co">#&gt; 2 &lt;NA&gt;   &lt;NA&gt;                    map05111 Biofilm formation… &lt;NA&gt;  &lt;NA&gt;         </span>
<span class="co">#&gt; 3 &lt;NA&gt;   &lt;NA&gt;                    map05111 Biofilm formation… &lt;NA&gt;  &lt;NA&gt;         </span>
<span class="co">#&gt; 4 M00878 Phenylacetate degradat… map05111 Biofilm formation… &lt;NA&gt;  &lt;NA&gt;         </span>
<span class="co">#&gt; 5 &lt;NA&gt;   &lt;NA&gt;                    map05111 Biofilm formation… Other Flagellar as…</span>
<span class="co">#&gt; 6 &lt;NA&gt;   &lt;NA&gt;                    map05111 Biofilm formation… &lt;NA&gt;  &lt;NA&gt;         </span>
<span class="co">#&gt; # … with 13 more variables: Detail_cycle &lt;chr&gt;, Genes &lt;chr&gt;,</span>
<span class="co">#&gt; #   Gene_description &lt;chr&gt;, Enzyme &lt;chr&gt;, KO &lt;chr&gt;, rbims_pathway &lt;chr&gt;,</span>
<span class="co">#&gt; #   rbims_sub_pathway &lt;chr&gt;, Bin_10 &lt;int&gt;, Bin_12 &lt;int&gt;, Bin_56 &lt;int&gt;,</span>
<span class="co">#&gt; #   Bin_113 &lt;int&gt;, Bin_1 &lt;int&gt;, Bin_2 &lt;int&gt;</span></code></pre></div>
<ul>
<li>Now, let’s create a plot of the presence and absence of the different KO associated with that pathway. Besides presence and absence, it is possible to plot abundance or the percentage of genes within certain pathways (See <a href="https://mirnavazquez.github.io/RbiMs/reference/plot_bubble.html">plot_bubble</a>, calc argument).</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_bubble.html">plot_bubble</a></span><span class="op">(</span>tibble_ko <span class="op">=</span> <span class="va">Biofilm_Vibrio_subset</span>,
            x_axis <span class="op">=</span> <span class="va">Bin_name</span>, 
            y_axis <span class="op">=</span> <span class="va">KO</span>,
            analysis<span class="op">=</span><span class="st">"KEGG"</span>,
            data_experiment <span class="op">=</span> <span class="va">metadata</span>,
            calc<span class="op">=</span><span class="st">"Binary"</span>,
            color_character <span class="op">=</span> <span class="va">Order</span>,
            range_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span> 
<span class="co">#&gt; Warning: Removed 66 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-9-1.png" width="672"></p>
<div id="order-axis" class="section level2">
<h2 class="hasAnchor">
<a href="#order-axis" class="anchor"></a>Order axis</h2>
<p>Let’s say that you want to order by bin names.</p>
<ul>
<li>Create a vector containing the order.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">order_taxa</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bin_1"</span>, <span class="st">"Bin_2"</span>, <span class="st">"Bin_10"</span>, <span class="st">"Bin_113"</span>, <span class="st">"Bin_12"</span>, <span class="st">"Bin_56"</span><span class="op">)</span></code></pre></div>
<ul>
<li>Now plot, using the order_bins argument.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_bubble.html">plot_bubble</a></span><span class="op">(</span>tibble_ko <span class="op">=</span> <span class="va">Biofilm_Vibrio_subset</span>,
            data_experiment <span class="op">=</span> <span class="va">metadata</span>,
            x_axis <span class="op">=</span> <span class="va">Bin_name</span>,
            y_axis <span class="op">=</span> <span class="va">Genes</span>,
            analysis<span class="op">=</span><span class="st">"KEGG"</span>,
            calc<span class="op">=</span><span class="st">"Binary"</span>,
            order_bins<span class="op">=</span><span class="va">order_taxa</span>,
            color_character<span class="op">=</span><span class="va">Genus</span>,
            range_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">6</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning: Removed 66 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
<ul>
<li>In the same way, we created a vector with the KEGG pathway ID of interest; we can create a vector of specific KOs IDs or KEGG modules.</li>
</ul>
<p>Here I will extract the information of some KO related to Carbon fixation metabolism.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Carbon_fixation</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"K01007"</span>, <span class="st">"K00626"</span>, <span class="st">"K01902"</span>, <span class="st">"K01595"</span>, <span class="st">"K01903"</span>, <span class="st">"K00170"</span>, <span class="st">"K00169"</span>, <span class="st">"K00171"</span>, <span class="st">"K00172"</span>, <span class="st">"K00241"</span><span class="op">)</span></code></pre></div>
<ul>
<li>Now, let’s extract the profile associated with that metabolic pathway.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="va">Carbon_fixation_subset</span><span class="op">&lt;-</span><span class="va">ko_bin_mapp</span><span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html">drop_na</a></span><span class="op">(</span><span class="va">KO</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/get_subset_pathway.html">get_subset_pathway</a></span><span class="op">(</span><span class="va">KO</span>, <span class="va">Carbon_fixation</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">Carbon_fixation_subset</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 19</span>
<span class="co">#&gt;   Module Module_description   Pathway  Pathway_descripti… Cycle  Pathway_cycle  </span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;                &lt;chr&gt;    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;          </span>
<span class="co">#&gt; 1 M00173 Reductive citrate c… map00010 Glycolysis / Gluc… Carbo… Reductive citr…</span>
<span class="co">#&gt; 2 M00173 Reductive citrate c… map00010 Glycolysis / Gluc… Carbo… Dicarboxylate-…</span>
<span class="co">#&gt; 3 M00374 Dicarboxylate-hydro… map00010 Glycolysis / Gluc… Carbo… Reductive citr…</span>
<span class="co">#&gt; 4 M00374 Dicarboxylate-hydro… map00010 Glycolysis / Gluc… Carbo… Dicarboxylate-…</span>
<span class="co">#&gt; 5 M00173 Reductive citrate c… map00620 Pyruvate metaboli… Carbo… Reductive citr…</span>
<span class="co">#&gt; 6 M00173 Reductive citrate c… map00620 Pyruvate metaboli… Carbo… Dicarboxylate-…</span>
<span class="co">#&gt; # … with 13 more variables: Detail_cycle &lt;chr&gt;, Genes &lt;chr&gt;,</span>
<span class="co">#&gt; #   Gene_description &lt;chr&gt;, Enzyme &lt;chr&gt;, KO &lt;chr&gt;, rbims_pathway &lt;chr&gt;,</span>
<span class="co">#&gt; #   rbims_sub_pathway &lt;chr&gt;, Bin_10 &lt;int&gt;, Bin_12 &lt;int&gt;, Bin_56 &lt;int&gt;,</span>
<span class="co">#&gt; #   Bin_113 &lt;int&gt;, Bin_1 &lt;int&gt;, Bin_2 &lt;int&gt;</span></code></pre></div>
<p>We can visualize the data with a <a href="https://mirnavazquez.github.io/RbiMs/reference/plot_heatmap.html">heatmap</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>tibble_ko<span class="op">=</span><span class="va">Carbon_fixation_subset</span>, 
             y_axis<span class="op">=</span><span class="va">Genes</span>,
             analysis <span class="op">=</span> <span class="st">"KEGG"</span>,
             calc<span class="op">=</span><span class="st">"Binary"</span><span class="op">)</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-15-1.png" width="672"></p>
</div>
<div id="the-calc-argument" class="section level2">
<h2 class="hasAnchor">
<a href="#the-calc-argument" class="anchor"></a>The calc argument</h2>
<p>In this example, we will use energy metabolism to explore the rest of the functions.</p>
<ul>
<li>Create a vector with the metabolism of interest.</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Other_energy</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Fermentation"</span>, <span class="st">"Carbon fixation"</span>, <span class="st">"Methane metabolism"</span>, 
                <span class="st">"Sulfur metabolism"</span>, <span class="st">"Nitrogen metabolism"</span><span class="op">)</span></code></pre></div>
<ul>
<li>Use <a href="https://mirnavazquez.github.io/RbiMs/reference/get_subset_pathway.html">get_subset_pathway</a> to subset the table using the cycles and the information of the energy metabolism.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="va">Energy_metabolisms</span><span class="op">&lt;-</span><span class="va">ko_bin_mapp</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html">drop_na</a></span><span class="op">(</span><span class="va">Cycle</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/get_subset_pathway.html">get_subset_pathway</a></span><span class="op">(</span><span class="va">Cycle</span>, <span class="va">Other_energy</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">Energy_metabolisms</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 19</span>
<span class="co">#&gt;   Module Module_description   Pathway  Pathway_descripti… Cycle  Pathway_cycle  </span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;                &lt;chr&gt;    &lt;chr&gt;              &lt;chr&gt;  &lt;chr&gt;          </span>
<span class="co">#&gt; 1 M00173 Reductive citrate c… map00010 Glycolysis / Gluc… Carbo… Reductive citr…</span>
<span class="co">#&gt; 2 M00173 Reductive citrate c… map00010 Glycolysis / Gluc… Carbo… Dicarboxylate-…</span>
<span class="co">#&gt; 3 M00374 Dicarboxylate-hydro… map00010 Glycolysis / Gluc… Carbo… Reductive citr…</span>
<span class="co">#&gt; 4 M00374 Dicarboxylate-hydro… map00010 Glycolysis / Gluc… Carbo… Dicarboxylate-…</span>
<span class="co">#&gt; 5 M00173 Reductive citrate c… map00620 Pyruvate metaboli… Carbo… Reductive citr…</span>
<span class="co">#&gt; 6 M00173 Reductive citrate c… map00620 Pyruvate metaboli… Carbo… Dicarboxylate-…</span>
<span class="co">#&gt; # … with 13 more variables: Detail_cycle &lt;chr&gt;, Genes &lt;chr&gt;,</span>
<span class="co">#&gt; #   Gene_description &lt;chr&gt;, Enzyme &lt;chr&gt;, KO &lt;chr&gt;, rbims_pathway &lt;chr&gt;,</span>
<span class="co">#&gt; #   rbims_sub_pathway &lt;chr&gt;, Bin_10 &lt;int&gt;, Bin_12 &lt;int&gt;, Bin_56 &lt;int&gt;,</span>
<span class="co">#&gt; #   Bin_113 &lt;int&gt;, Bin_1 &lt;int&gt;, Bin_2 &lt;int&gt;</span></code></pre></div>
<ul>
<li>Plot the information using the <a href="https://mirnavazquez.github.io/RbiMs/reference/plot_heatmap.html">plot_heatmap</a> function. The argument order_y will order the rows according to a metabolic feature; in this case, we order the pathways_cycle according to cycle.</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>tibble_ko<span class="op">=</span><span class="va">Energy_metabolisms</span>, 
             y_axis<span class="op">=</span><span class="va">Pathway_cycle</span>,
             order_y <span class="op">=</span> <span class="va">Cycle</span>,
             analysis <span class="op">=</span> <span class="st">"KEGG"</span>,
             calc<span class="op">=</span><span class="st">"Percentage"</span><span class="op">)</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-19-1.png" width="672"></p>
<ul>
<li>The argument order_x will order the rows according to a metadata feature; in this case, we order the bins according to sample site.</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>tibble_ko<span class="op">=</span><span class="va">Energy_metabolisms</span>, 
             y_axis<span class="op">=</span><span class="va">Pathway_cycle</span>,
             data_experiment<span class="op">=</span><span class="va">metadata</span>,
             order_x <span class="op">=</span> <span class="va">Sample_site</span>,
             analysis <span class="op">=</span> <span class="st">"KEGG"</span>,
             calc<span class="op">=</span><span class="st">"Percentage"</span><span class="op">)</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-20-1.png" width="672"></p>
<ul>
<li>The split argument allows dividing the rows according to a specific value of the metadata.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>tibble_ko<span class="op">=</span><span class="va">Energy_metabolisms</span>, 
             y_axis<span class="op">=</span><span class="va">Pathway_cycle</span>,
             order_y <span class="op">=</span> <span class="va">Cycle</span>,
             split_y <span class="op">=</span> <span class="cn">TRUE</span>,
             analysis <span class="op">=</span> <span class="st">"KEGG"</span>,
             calc<span class="op">=</span><span class="st">"Percentage"</span><span class="op">)</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-21-1.png" width="672"></p>
<ul>
<li>The order_x argument allows you to add annotation info from the metadata for the columns.</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>tibble_ko<span class="op">=</span><span class="va">Energy_metabolisms</span>,
             data_experiment <span class="op">=</span> <span class="va">metadata</span>,
             y_axis<span class="op">=</span><span class="va">Pathway_cycle</span>,
             order_y <span class="op">=</span> <span class="va">Cycle</span>,
             order_x <span class="op">=</span> <span class="va">Clades</span>,
             split_y <span class="op">=</span> <span class="cn">TRUE</span>,
             analysis <span class="op">=</span> <span class="st">"KEGG"</span>,
             calc<span class="op">=</span><span class="st">"Percentage"</span><span class="op">)</span></code></pre></div>
<p><img src="Explore_KEGG_profile_files/figure-html/unnamed-chunk-22-1.png" width="672"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mirna Vázquez-Rosas-Landa.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
